### PR TITLE
Deprecate "pytestmark" in favor of "pytest_marks"

### LINKED ISSUE #816: pytestmark vs pytest_marks
Currently you can use `pytest_plugins` (accepts a single string or a list of strings) to explicitly define plugin dependencies in a module, and `pytestmark` to set one or more marks into all tests in a module or class. IMO it is unfortunate that they follow a completely different naming standard, as this has tripped me more than once.

I propose that we introduce `pytest_marks` with the exact same semantics and deprecate `pytestmark` in the documentation. 

Any thoughts?

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #816

### CODE REVIEW

File: doc/en/example/markers.txt [hunk 1]
```diff
@@ -242,28 +242,34 @@ This is equivalent to directly applying the decorator to the
...
-    pytestmark = pytest.mark.webtest
+    pytest_marks = pytest.mark.webtest
 
 in which case it will be applied to all functions and
 methods defined in the module.
 
+.. deprecated:: 2.8
+    Previously the variable was named ``pytestmark``, but for consistency
+    with other pytest variables it was renamed to ``pytest_marks``.
+    ``pytestmark`` still works, but users are encourage to use ``pytest_marks``
```
Comment (The-Compiler) [inline doc/en/example/markers.txt]: encourage -> encouraged

File: testing/test_mark.py [hunk 2]
```diff
@@ -263,42 +270,44 @@ def test_hello():
...
                 def test_func(self):
                     assert TestClass.test_func.hello
                     assert TestClass.test_func.world
-        """)
+        """ % var_name)
         keywords = item.keywords
         assert 'hello' in keywords
         assert 'world' in keywords
 
-    @pytest.mark.skipif("sys.version_info < (2,6)")
```
Comment (The-Compiler) [inline testing/test_mark.py]: Is it intended that you removed this?

File: doc/en/example/markers.txt (hunk 1 above)
Comment (nicoddemus) [inline doc/en/example/markers.txt]: Fixed, thanks!

File: testing/test_mark.py (hunk 2 above)
Comment (nicoddemus) [inline testing/test_mark.py]: Yes, `pytest` has dropped Python 2.5 in release `2.5.0`. From the `CHANGELOG`:

> dropped python2.5 from automated release testing of pytest itself which means it's probably going to break soon (but still works with this release we believe). 

So I decided to remove the line noise. :blush:

**Comment by hpk42** [comment]
i am not sure it's worth the trouble to go for this naming change TBH.  If we ever drop ``pytestmark`` lots of test suites are going to fail  (not matter how well you communicate deprecation). 

in any case, if we do a deprecation then a warning should be generated for each module/class which uses it.

**Comment by nicoddemus** [comment]
No problem @hpk42. I don't feel to strongly about it either, I opened the issue mostly to see what others think, and decided to open the PR to foster discussion.

I propose we drop this then! :smile:

**Comment by hpk42** [comment]
i'd say keep it in your private repo.  

FWIW I have a patch pending which majorly improves marking (since 1-2 years already, alas).  When i get around to do a proper PR, it might make sense to discuss your change as well.  If others disagree and rather would like to get your PR integrated now, i am fine with including it, however.

**Comment by nicoddemus** [comment]
No problem, I will leave this around on my fork then.

Let's wait for your mark changes, we might decide to go in another route altogether and then we would have to keep supporting the new `pytest_marks` in addition to `pytestmark`.

Cheers

**Pull request closed without merging**